compatibility for current ArrayFire and current OpenCV

meetnick opened the pull request:
With the respect of its creator developer, I send this pull request in order to make this compatible with current versions of ArrayFire and OpenCV (tested on 3.7.0 and 3.4.3, respectively). This pull request includes the following in order to accomplish:
- Stick with std:: library for **std::vector** and **std::string** (in order to provide better compatibility and prevent undesired namespace propagation).
- corrects the name of af/utils.h to **af/util.h** which is how's officially called in ArrayFire.
- updates deprecated colorspace to current **colorSpace**.
- changes af_cspace_t af_gray and af_rgb to **AF_GRAY** and **AF_RGB**, respectivelly (maybe TYPO?)
- Fixes the possibility of crash when using single-dimensional array. On this case, it should consider the number of elements instead ask for nonexistent dim(ndims - 2), which gives segmentation fault error.
- With respect to its creator, I removed "Mat stats" functions definition and declarations. I definitely don't know where _top and _stats came from.

It also solves issue #1

System environment:
*buntu 18.04.1 LTS
OpenCV 3.4.3 
ArrayFire v3.7.0 (CPU|OpenCL, 64-bit Linux, build ef5183d7)

meetnick committed:
make it compatible for current ArrayFire (3.7.0) and current OpenCV (3.4.3)

mcclanahoochie commented on af_cv.cpp:
```diff
@@ -166,21 +166,25 @@ void array_to_mat(const array& input_, cv::Mat& output, int type, bool transpose
     } else if (type == CV_32S) {
         int* data = output.ptr<int>(0);
-        input.as(af::s32).host((void*)data);
+        input.as(s32).host((void*)data);
```
if af:: no longer needed?

mcclanahoochie replied:
ah, i see 'using af' is in header, oops
up to you if you prefer to remove af:: or keep

meetnick replied:
actually ArrayFire data types aren't directly under af::, they are under af::dtype::
```suggestion
        input.as(af::dtype::s32).host((void*)data);
```

mcclanahoochie commented:
[niccolatorres](/niccolatorres) thanks for the update!!

mcclanahoochie reviewed (changes requested):
first off, i really appreciate the update!!! 

i'd prefer to keep the af:: namespace around, since you didn't remove af:: from af::exception.. 
so,
completely remove af:: everywhere because of the 'using namespace af' , 
or revert the "af::" changes.
thanks again!

meetnick committed:
remove OpenCV and ArrayFire namespace usage within header to avoid undesirable namespace propagation

meetnick reviewed (commented):
- removed OpenCV (cv) and ArrayFyre (af) namespaces from header to avoid undesirable namespace propagation.
- **cv::** and **af::** is now explicitly used
- **af::dtype::** is used to reference arrayfire data types

meetnick commented:
> first off, i really appreciate the update!!!
> 
> i'd prefer to keep the af:: namespace around, since you didn't remove af:: from af::exception..
> so,
> completely remove af:: everywhere because of the 'using namespace af' ,
> or revert the "af::" changes.
> thanks again!

I much appreciate to contribute to this project. I indeed agree af:: and even cv:: namespaces should be used explicitly inside to avoid namespace undesirable propagation. I removed af:: from data types because compiler was raising "af:: has no member u8/f64/s32/etc.." Now I fixed it to the correct namespace for data types, which is **af::dtype::**

Please consider the new commit with these changes requested.

mcclanahoochie reviewed (approved):
LGTM. Thank you!
